cli/azd/cmd/deploy.go

```diff
@@ -177,45 +180,32 @@ func (d *deployAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 			continue
 		}
 
-		deployAndReportProgress := func(ctx context.Context, showProgress func(string)) error {
-			result, progress := svc.Deploy(ctx, d.azdCtx)
+		stepMessage := fmt.Sprintf("Deploying service %s", svc.Config.Name)
+		d.console.ShowSpinner(ctx, stepMessage, input.Step)
+		result, progress := svc.Deploy(ctx, d.azdCtx)
 
-			// Report any progress
-			go func() {
-				for message := range progress {
-					showProgress(fmt.Sprintf("- %s...", message))
-				}
-			}()
-
-			response := <-result
-			if response.Error != nil {
-				return fmt.Errorf("deploying service: %w", response.Error)
+		// Report any progress to logs only. Changes for the console are managed by the console object.
```

Comment:
right, I have updated the comment to be `routine` instead, to avoid the confusion.

Just for the record, here's a nice document about the `go routines`: https://www.alibabacloud.com/forum/read-916

Even knowing that `goroutines != os threads`, the chances of using multiple os threads by the go-runtime are higher as the number of goroutines increases.  I prefer to minimize the usage of goroutines only for cases where they are really necessary.  

For example, azd uses around ~7 routines while running provision:

![image](https://user-images.githubusercontent.com/24213737/207760111-8ba5852b-d81d-49c0-8e91-f9606c9eb9fa.png)

Some of those routines are switching concurrently within the go-runtime, and there might also be some os kernel threads linked to the one or more than one groups of go-routines (sharing one HW CPU core) in parallel.